homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -246,6 +268,53 @@
     ATTR_EXTRA_FEATURES: 'extra_features',
 }
 
+AVAILABLE_ATTRIBUTES_FAN = {
+    ATTR_ANGLE: 'angle',
+    ATTR_SPEED: 'speed',
+    ATTR_DELAY_OFF_COUNTDOWN: 'delay_off_countdown',
+
+    ATTR_AC_POWER: 'ac_power',
+    ATTR_OSCILLATE: 'oscillate',
+    ATTR_DIRECT_SPEED: 'direct_speed',
+    ATTR_NATURAL_SPEED: 'natural_speed',
+    ATTR_CHILD_LOCK: 'child_lock',
+    ATTR_BUZZER: 'buzzer',
+    ATTR_LED_BRIGHTNESS: 'led_brightness',
+    ATTR_USE_TIME: 'use_time',
+
+    # Additional properties of version 2 and 3
+    ATTR_TEMPERATURE: 'temperature',
+    ATTR_HUMIDITY: 'humidity',
+    ATTR_BATTERY: 'battery',
+    ATTR_BATTERY_CHARGE: 'battery_charge',
+    ATTR_BUTTON_PRESSED: 'button_pressed',
+
+    # Additional properties of version 2
+    ATTR_LED: 'led',
+    ATTR_BATTERY_STATE: 'battery_state',
+}
+
+FAN_SPEED_LEVEL1 = 'Level 1'
+FAN_SPEED_LEVEL2 = 'Level 2'
+FAN_SPEED_LEVEL3 = 'Level 3'
+FAN_SPEED_LEVEL4 = 'Level 4'
+
+FAN_SPEED_LIST = {
```

Comment:
These keys aren't part of the base fan component so aren't really allowed. I know this platform already breaks this rule. Should we continue breaking it?

Reply:
@balloob?

Reply:
@balloob Ping? ;-) Feel free. At the moment the implementation is in sync with the MiHome app.

Reply:
Yo, yeah we have our own list that you should map the xiaomi ones to.

Reply:
We create an abstraction such that we can provide services on top like Google Assistant or other UIs integrating with us.